WooCommerce/WooCommerceTests/ViewRelated/Products/Add Product/AddProductWithAI/ProductDetailPreviewViewModelTests.swift

```diff
@@ -601,6 +601,143 @@ final class ProductDetailPreviewViewModelTests: XCTestCase {
         XCTAssertEqual(generatedProduct.tags.map { $0.tagID }, [0, 0])
     }
 
+    func test_generateProductDetails_switches_to_given_productName_if_AIProduct_has_empty_name() async throws {
+        // Given
+        let product = AIProduct.fake().copy(name: "", description: "Test description")
+        let viewModel = ProductDetailPreviewViewModel(siteID: sampleSiteID,
+                                                      productName: "Pen",
+                                                      productDescription: nil,
+                                                      productFeatures: "Ballpoint, Blue ink, ABS plastic",
+                                                      weightUnit: "kg",
+                                                      dimensionUnit: "m",
+                                                      stores: stores,
+                                                      storageManager: storage,
+                                                      onProductCreated: { _ in })
+
+        mockProductActions(aiGeneratedProductResult: .success(product))
+        mockProductTagActions()
+        mockProductCategoryActions()
+
+        // When
+        await viewModel.generateProductDetails()
+
+        // Then
+        let generatedProduct = try XCTUnwrap(viewModel.generatedProduct)
+        XCTAssertEqual(generatedProduct.name, "Pen")
+    }
+
+    // MARK: Short description view
+
+    func test_short_description_view_is_shown_if_shortDescription_is_not_empty() async throws {
```

Comment:
Nit: `throws` is redundant here (similar to other tests below).

Reply:
Removed in https://github.com/woocommerce/woocommerce-ios/pull/11853/commits/69fdf0a1bf633f23e3a8228851dec8b50708b063